js/runner_test.go

```diff
@@ -268,6 +268,161 @@ func TestSetupTeardown(t *testing.T) {
 	}
 }
 
+func TestSetupDataIsolation(t *testing.T) {
+	r1, err := New(&lib.SourceData{
+		Filename: "/script.js",
+		Data: []byte(`
+			export let options = {
+				setupTimeout: "10s",
+				teardownTimeout: "10s",
+			};
+
+			export function setup() {
+				return { v: 1 };
+			}
+			export function teardown(data) {
+				if (data.v != 1) {
+					throw new Error("teardown: wrong data: " + data.v)
+				}
+				data.v = 2
+			}
+			export default function(data) {
+				if (data.v != 1) {
+					throw new Error("default: wrong data: " + JSON.stringify(data))
+				}
+				data.v = 2
+			}
+		`),
+	}, afero.NewMemMapFs(), lib.RuntimeOptions{})
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	r2, err := NewFromArchive(r1.MakeArchive(), lib.RuntimeOptions{})
```

Comment:
Same note about archives in this test as in the other - not sure testing both archives and scripts would help us catch any errors in this context.